java-squid/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java

```diff
@@ -154,18 +160,61 @@ public File getFile() {
     return complexityVisitor.scan(enclosingClass, methodTree);
   }
 
-  public void reportSEIssue(Class<? extends SECheck> check, Tree tree, String message) {
+  public void reportSEIssue(Class<? extends SECheck> check, Tree tree, String message, List<Location> secondary) {
     if (!seIssues.containsKey(check)) {
-      seIssues.put(check, LinkedHashMultimap.<Tree, String>create());
+      seIssues.put(check, LinkedHashMultimap.<Tree, SEIssue>create());
     }
-    seIssues.get(check).put(tree, message);
+    seIssues.get(check).put(tree, new SEIssue(tree, message, secondary));
   }
 
-  public Multimap<Tree, String> getSEIssues(Class<? extends SECheck> check) {
+  public Multimap<Tree, SEIssue> getSEIssues(Class<? extends SECheck> check) {
     if (seIssues.containsKey(check)) {
       return seIssues.get(check);
     } else {
       return ImmutableMultimap.of();
     }
   }
+
+  public static class SEIssue {
+    private final Tree tree;
+    private final String message;
+    private final List<Location> secondary;
+
+    public SEIssue(Tree tree, String message, List<Location> secondary) {
+      this.tree = tree;
+      this.message = message;
+      this.secondary = secondary;
+    }
+
+    public Tree getTree() {
+      return tree;
+    }
+
+    public String getMessage() {
+      return message;
+    }
+
+    public List<Location> getSecondary() {
```

Comment:
Maybe the class SEIssue could generate the message passed to the method CheckerContext#reportIssue instead of adding a new parameter.